src/core/lib/layer.js

```diff
@@ -116,6 +116,10 @@ export default class Layer {
     return this.state.stats;
   }
 
+  get isPickable() {
+    return this.props.pickable;
```

Comment:
visible && pickable ?

Reply:
>  get isPickable() {

1) In general I feel that getters (the `get` keyword) should be used sparingly and with rigorous consistency, as they can cause as much confusion as they can help (looks like a member/field but is actually a function and magically changes value). 

Let's drop the `get` for now unless we articulate a strategy.

2) Alternative is to drop the `get` and just define an `isPickable` method. 

Do we need another method just to wrap `layer.props.pickable`?  If there was more complex logic to determine "pickability" it would seem more strongly justified. I am certainly open to the idea, but if we introduce such a method/accessor we should really go through the code (possibly in another PR) and make sure it is used everywhere, otherwise we are just making the code more inconsistent. 

We might also consider if there are other similar accessors we should add, `isVisible()`, etc.

Reply:
changed `getter` to member function, added `visible` check. Apparently we were going through all picking logic even when layers are not visible.

Reply:
Question (not for this PR): How should handle a layer that is visible but not pickable. Should we render it into the picking buffer using null color to make sure we can't pick what is behind it? It would be a small extension to the picking module.

Right now I believe that users can pick "through" occluding, non-pickable layers.